yaguzmang/ghg-emission-calculation-tool
# Feature/next/add main emissions form view

Closes #56 


## Description

Adding the main emissions form, following the Figma designs and CRUD operations with validation and translated error messages.
This PR also includes:
- Changes in the UI in the dashboard.
- Improved responsiveness. 

## Before testing:

- Add the following translation keys and values (translations also included at the end of the PR without markdown formatting):
  - [ ] `dashboard.home`: `Home`
  - [ ] `dashboard.form.emissionsSummary.showResults`: `Show results based on activity`
  - [ ] `dashboard.form.emissionEntry.addNewCategoryEmission`: `Add new {{categoryTitle}} emissions`
  - [ ] `forms.saveChanges`: `Save changes`
  - [ ] `dashboard.form.emissionEntry.deleteTooltip`: `Delete`
  - [ ] `dashboard.form.emissionEntry.deleteConfirm`: `Yes, delete emission`
  - [ ] `dashboard.form.emissionEntry.organizationUnit.error.select`: `Please select an organization unit.`
  - [ ] `dashboard.form.emissionEntry.reportingPeriod.error.select`: `Please select a reporting period.`
  - [ ] `dashboard.form.emissionEntry.emissionSource.error.select`: `Please select an emission source.`
  - [ ] `dashboard.form.emissionEntry.quantity.error.valid`: `Please input a valid quantity.`
  - [ ] `dashboard.form.emissionEntry.tier.error.select`: `Please select a tier.`
  - [ ] `dashboard.form.emissionEntry.emissionFactors.value.error.positive`: `Please provide a positive value for the custom emission factor.`
  - [ ] `dashboard.form.emissionEntry.emissionFactors.source.error.input`: `Please input a source.`
## How to test

- In /frontend:
- In /backend:
  - [ ] `npm run develop` 

Check that the app works: 
  - [ ] Sign in (you should be redirected to `/dashboard`).
  - [ ] Click on an emission category to go to its emissions form.
  - [ ] If the category already has emissions, they should be listed and you should be able to edit/delete them.
  - [ ] Add a new emission and check that the summary on the right is correct.


![Large GIF (1426x930)](https://github.com/yaguzmang/ghg-emission-calculation-tool/assets/41337901/6d02c13f-9d0e-4ff8-a6c3-978c0c039dac)

Translations without formatting:
dashboard.home
Home

dashboard.form.emissionsSummary.showResults
Show results based on activity

dashboard.form.emissionEntry.addNewCategoryEmission
Add new {{categoryTitle}} emissions

forms.saveChanges
Save changes

dashboard.form.emissionEntry.deleteTooltip
Delete

dashboard.form.emissionEntry.deleteConfirm
Yes, delete emission

dashboard.form.emissionEntry.organizationUnit.error.select
Please select an organization unit.

dashboard.form.emissionEntry.reportingPeriod.error.select
Please select a reporting period.

dashboard.form.emissionEntry.emissionSource.error.select
Please select an emission source.

dashboard.form.emissionEntry.quantity.error.valid
Please input a valid quantity.

dashboard.form.emissionEntry.tier.error.select
Please select a tier.

dashboard.form.emissionEntry.emissionFactors.value.error.positive
Please provide a positive value for the custom emission factor.

dashboard.form.emissionEntry.emissionFactors.source.error.input
'Please input a source.'
